### PR TITLE
Add forceSynch option which will autosave to textarea.

### LINKED ISSUE #193: Have SimpleMDE keep the <textarea> updated with new content
Would be great if it was! Also would be nice if SimpleMDE could listen for changes in the textarea.

### PR DESCRIPTION
**Pull request by cbadke**
Since this is built on CodeMirror I opted to use CodeMirror's save
behaviour rather than pulling the value and stomping on the textarea's
innerHtml.

See this [jsfiddle](https://jsfiddle.net/njjeemye/3/) to see it in action.

Fixes #193

### CODE REVIEW

File: src/js/simplemde.js [hunk 1]
```diff
@@ -1484,6 +1484,13 @@ SimpleMDE.prototype.render = function(el) {
 		placeholder: options.placeholder || el.getAttribute("placeholder") || ""
 	});
 
+	if(options.forceSynch !== false) {
```
Comment (WesCossick) [inline src/js/simplemde.js]: Change to `=== true`

File: README.md [hunk 2]
```diff
@@ -78,6 +78,7 @@ simplemde.value("This text will appear in the editor");
   - **code** Can be set to  ```` ``` ```` or `~~~`.  Defaults to ```` ``` ````.
   - **italic** Can be set to `*` or `_`. Defaults to `*`.
 - **element**: The DOM element for the textarea to use. Defaults to the first textarea on the page.
+- **forceSynch**: If set to `true`, force changes made in editor to be saved back to original textarea.
```
Comment (WesCossick) [inline README.md]: Let's rename the option to `forceSync` and set the description as:

> If set to `true`, force text changes made in SimpleMDE to be immediately stored in original textarea. Defaults to `false`.

**Comment by WesCossick** [comment]
I just made a few comments in the code about some changes I'd prefer before merging it in.

**Comment by cbadke** [comment]
[Here](https://jsfiddle.net/njjeemye/4/) is the jsFiddle representing new naming.

**Comment by WesCossick** [comment]
Great! One last change, can you demo the non-default option in the block of code in the README that demonstrates all options? For this instance, you'd need to add `forceSync: true,` in the appropriate alphabetical location.

**Comment by WesCossick** [comment]
Merged in #264. Had to make a commit to eliminate conflicts. Thanks for the contribution!

**Comment by potasiyam** [comment]
Thanks for this. Now textarea with required attribute will not break.

**Pull request closed without merging**